prov/gni/src/gnix_hashtable.c

```diff
@@ -715,9 +724,6 @@ int _gnix_ht_remove(gnix_hashtable_t *ht, gnix_ht_key_t key)
 
 void *_gnix_ht_lookup(gnix_hashtable_t *ht, gnix_ht_key_t key)
 {
-	if (ht->ht_state != GNIX_HT_STATE_READY)
-		return NULL;
-
```

Comment:
I'm not certain that we'd need that. It should immediately segfault when it attempts to dereference a bucket because memory would not be allocated. I'd be fine with adding an assert though.